tasks/interview-corejs.md

```diff
@@ -1,60 +1,278 @@
+  - Understand connection between function and lexical environment
+
+- #### Object Oriented Programming
+
+  - `new` keyword
+  - Understand how `new` keyword works
+  - Function constructor
+  - Know function constructor concept
+  - Know constructor function pattern
+  - Able to create constructor functions
+  - Public, private, static members
+  - Know how to create public members
+  - Know how to create private members
+  - Know how to create static members
+  - Understand OOP emulation patterns and conventions
+
+- #### Prototypal Inheritance Basics
+
+  - `__proto__` property
+  - Understand `__proto__` object property
+  - Able to use [Object.create] and define `__proto__` explicitly
+  - Able to set / get object prototype
+  - `prototype` property
+  - Know function `prototype` property
+  - Understand dependency between function constructor `prototype` and instance `__proto__`
+  - Able to create 'class' methods using function `prototype` property
+
+- #### ECMAScript Classes
+
+  - Class declaration
+  - Know `class` declaration syntax
+  - Understand difference between `class` and `function constructor`
+  - Getter/setter
+  - What does `super()` do and where we have to use it?
+
+- #### ECMAScript Data Types & Expressions
+
+  - Object `keys/values`
+  - Object calculated props
+  - `Set/Map` data types
+  - `WeakSet/WeakMap` data types
+
+- #### JavaScript Errors
+
+  - `try..catch` statement
+  - Know how to handle errors
+  - Custom errors
+
+- #### ECMAScript Advanced
+
+  - Garbage collector (concept)
+  - Promises
+  - Promise states
+  - Promise Chaining
+  - Promise Methods
+  - Be able to compare promise and callback patterns
+  - Be able to handle errors in promises
+  - event loop
+  - async/await
+
+### JavaScript in Browser:
+
+- #### Global object window
+
+  - Location
+  - Know browser location structure
+  - History API (Global object window)
+  - Know browser History APIconcept
+  - Be able to navigate within browser history
+  - Be able to use history state
+  - Navigator
+  - Know how to parse user agent
+  - Know how to discover client platform, browser
+  - Document
+  - Cookies
+
+- #### Page Lifecycle
+
+  - Parsing
+  - Reflow
+  - Repaint
+
+- #### Events Basics
+
+  - Event Phases
+  - Be able to explain difference between capturing and bubbling
+  - Event Listeners
+  - DOM Events
+  - Know Event concept
+  - Know basic Event types
+  - Mouse / Keyboard Events
+  - Form / Input Events
+
+- #### Events Propagation / Preventing
+
+  - Know Event propagation cycle
+  - Know how to stop Event propagation
+  - Know how to prevent Event default browser behavior
+  - Delegating
+  - Understand Event delegating concept
+  - Understand Event delegating benefits and drawbacks
+
+- #### Timers
+
+  - `setTimeout`
+  - `setInterval`
+  - `clearTimeout`
+  - `requestAnimationFrame`
+  - Be able to explain difference between `setTimeout` and `requestAnimationFrame`
```

Comment:
Add
- Be able to explain if `setTimeout (callback, 1500)` will execute **exactly** 1500 milliseconds later